platform/ui/src/components/ViewportOverlay/index.js

```diff
@@ -0,0 +1,117 @@
+const generateFromConfig = (config) => {
+  const { topLeft, topRight, bottomLeft, bottomRight, itemGenerator } = config;
+
+  return (props) => {
+    const { imageId } = props;
+    const topLeftClass = 'top-viewport left-viewport';
+    const topRightClass = 'top-viewport right-viewport-scrollbar';
+    const bottomRightClass = 'bottom-viewport right-viewport-scrollbar';
+    const bottomLeftClass = 'bottom-viewport left-viewport';
+    const overlay = 'absolute pointer-events-none';
+
+    if (!imageId) {
+      return null;
+    }
+
+    return (
+      <div className="text-primary-light">
+        <div
+          data-cy={'viewport-overlay-top-left'}
+          className={classnames(overlay, topLeftClass)}
+        >
+          {listComponentGenerator({ ...props, list: topLeft, itemGenerator })}
+        </div>
+        <div
+          data-cy={'viewport-overlay-top-right'}
+          className={classnames(overlay, topRightClass)}
+        >
+          {listComponentGenerator({ ...props, list: topRight, itemGenerator })}
+        </div>
+        <div
+          data-cy={'viewport-overlay-bottom-right'}
+          className={classnames(overlay, bottomRightClass)}
+        >
+          {listComponentGenerator({ ...props, list: bottomRight, itemGenerator })}
+        </div>
+        <div
+          data-cy={'viewport-overlay-bottom-left'}
+          className={classnames(overlay, bottomLeftClass)}
+        >
+          {listComponentGenerator({ ...props, list: bottomLeft, itemGenerator })}
+        </div>
+      </div>
+    );
+  };
+};
+
+const { ViewportOverlay } = ConfigPoint.register({
+  ViewportOverlay: {
+    configBase: {
+      topLeft: [
+        {
+          id: 'zoom',
+          title: 'Zoom:',
+          condition: props => props.activeTools && props.activeTools.includes("Zoom"),
+          value: props => props.scale && (props.scale.toFixed(2) + "x"),
+        },
+        {
+          id: 'wwwc',
+          condition: props => props.activeTools && props.activeTools.includes('Wwwc'),
+          contents: props => ([
+            { className: 'mr-1', value: "W:" },
+            { className: 'ml-1 mr-2 font-light', value: props.windowWidth.toFixed(0) },
+            { className: 'mr-1', value: "L:" },
+            { className: 'ml-1 font-light', value: props.windowCenter.toFixed(0) },
+          ]),
+        },
+      ],
+      topRight: [
+        {
+          id: 'stackSize',
+          // An example of how to run this with a dynamic, safe function
+          condition: { configOperation: 'safe', value: 'stackSize > 1 && image' },
+          title: "I:",
+          value: props => `${props.image.InstanceNumber} (${props.imageIndex}/${props.stackSize})`,
+        },
+      ],
+      bottomLeft: [
+      ],
+      bottomRight: [
+      ],
+      itemGenerator: props => {
+        const { item } = props;
+        const { title, value: valueFunc, condition, contents } = item;
+        if (condition && !condition(props)) return null;
```

Comment:
The condition here might be a string, how you're handling the safe function which is a string?

Reply:
The point of the safe function is that you can define the function using a string, so it can come from a format like JSON, but it gets auto-converted into a function that is safe for execution because the context is tightly controlled, only allowing values from the single argument of hte function to be used.

Reply:
Do I have to change something on my end? I'm getting: condition is not a function when using these changes.

Reply:
You need to do a yarn install probably - then check that node_modules/config-point/package.json has version 0.4.1 in it.